mlir/include/mlir/Dialect/LLVMIR/GENXOps.td

```diff
@@ -338,7 +338,64 @@ def GENX_AtomicRMWOp : GENX_Op<"atomic.rmw", [
 //===----------------------------------------------------------------------===//
 // Joint Matrix operations
 //===----------------------------------------------------------------------===//
+
 def GENX_MatrixElemType : AnyTypeOf<[AnyI8, AnyI16, AnyI32, F32, F16, BF16]>;
+
+def GENX_MatrixDPASOp : GENX_Op<"matrix.dpas">,
+  Results<(outs FixedVectorOf<[GENX_MatrixElemType]>:$d)>,
+  Arguments<(ins 
+    FixedVectorOf<[GENX_MatrixElemType]>:$c,
+    FixedVectorOf<[GENX_MatrixElemType]>:$a,
+    FixedVectorOf<[GENX_MatrixElemType]>:$b,
+    I32Attr:$pa,
+    I32Attr:$pb,
+    I32Attr:$sd,
+    I32Attr:$rc,
+    I1Attr:$isDpasw
+  )> {
+
+  let summary = "GEN matrix multiply-add";
+
+  string baseDescription = [{
+    The 'genx.matrix.dpas' operation is a matrix multiply-add operation as follows:
+
+       D = C + A x B
+
+       where
+          D : MxN
+          C : MxN
+          A : MxK
+          B : KxN
+
+          M : repeat count ($rc)
+          N : fixed execution size, either 8 or 16
+          K : depth * OPS_PER_CHAN
+              OPS_PER_CHAN
+                 1 : for TF32
+                 2 : for 16-bit precision(BF, HF)
+                 4 : for 8-bit precision (FP8, UB, B)
+                 8 : for less-then 8 bit precision (U4/S4, U2/S2).
+
+              If depth is 8, K would be 8, 16, 32, or 64 (based on OPS_PER_CHAN).
+
+    $a, $b, $c, $d - matrix A, B, C, D, respectively
+    $pa, $pb - precision of matrix A and B resepectively
+    $sd - systolic depth
+    $rc - repect count
+    $isDpasw - is dpas wide instruction
+  }];
+
+  string llvmBuilder = [{
+    $d = createGenISADPAS(op, builder, moduleTranslation);
```

Comment:
I think passing op is cleaner than passing each argument. Will change `GENX_Matrix2DBlockLoadOp` to do the same, if no one disagrees.

Reply:
I agree.  Thanks.

Reply:
https://github.com/pengtu/llvm-project/pull/30